### PR TITLE
Open Scmp file in Schema Compare extension

### LINKED ISSUE #15880: Open Schema Compare from .scmp-file in file explorer
In our database projects we have several scmp-files with pre-configured schema compares. 
As now I can use the files by opening the _Schema Compare_ from the context-menu on the connection, and then click _Open .scmp-file_.

If I try to open the file from the file explorer or the project explorer I only get the underlaying XML-file, which mostly often isn't very useful.
It would be great to open a Schema Compare tab with the selected file loaded.

An alternative would be a context menu with the option _Open in Schema Compare_ or similar.

### PR DESCRIPTION
**Pull request by SakshiS-harma**
This PR fixes #15880.

For .scmp files present in file explorer will now have an extra option to "Open in Schema Compare" in context menu.

image

scmpFileOpen

### CODE REVIEW

File: extensions/schema-compare/src/schemaCompareMainWindow.ts [hunk 1]
```diff
@@ -1089,6 +1089,13 @@ export class SchemaCompareMainWindow {
 		}
 
 		let fileUri = fileUris[0];
+		this.openScmpFile(fileUri, true);
+
+	}
+	public async openScmpFile(fileUri: vscode.Uri, callFromWithinSC: boolean = false): Promise<void> {
+		if (!callFromWithinSC) {
+			await this.launch(undefined, undefined, false, undefined);
```
Comment (kisantia) [inline extensions/schema-compare/src/schemaCompareMainWindow.ts:1097]: does this open a new schema compare window?

File: extensions/schema-compare/src/schemaCompareMainWindow.ts [hunk 2]
```diff
@@ -1089,6 +1089,13 @@ export class SchemaCompareMainWindow {
 		}
 
 		let fileUri = fileUris[0];
+		this.openScmpFile(fileUri, true);
+
+	}
+	public async openScmpFile(fileUri: vscode.Uri, callFromWithinSC: boolean = false): Promise<void> {
```
Comment (kisantia) [inline extensions/schema-compare/src/schemaCompareMainWindow.ts:1095]: nit: add blank line before this

Comment (kisantia) [inline extensions/schema-compare/src/schemaCompareMainWindow.ts:1095]: add js doc comments for this function

**Review by kisantia — COMMENTED** [review]
can you also add a gif showing that shows the schema compare window opening with the source and target populated?

File: extensions/schema-compare/src/schemaCompareMainWindow.ts (hunk 1 above)
Comment (SakshiS-harma) [reply extensions/schema-compare/src/schemaCompareMainWindow.ts:1097]: Yes, this does open a new SC window. And later in following lines, it reads the file and populates the source and target information appropriately.

Comment (kisantia) [reply extensions/schema-compare/src/schemaCompareMainWindow.ts:1097]: ok cool can you add a comment about that and why it's needed now that there's this new entry point?

Comment (SakshiS-harma) [reply extensions/schema-compare/src/schemaCompareMainWindow.ts:1097]: Sure, will do.

**Comment by SakshiS-harma** [comment]
> can you also add a gif showing that shows the schema compare window opening with the source and target populated?

Just updated the PR with a gif.

File: extensions/schema-compare/src/schemaCompareMainWindow.ts [hunk 3]
```diff
@@ -1089,6 +1089,18 @@ export class SchemaCompareMainWindow {
...
+	}
+
+	/**
+	 * Primary functional entrypoint for opening the schema comparison window with the scmp file Uri provided.
+	 * @param fileUri .scmp file URI to open Schema Compare extension with
+	 * @param callFromWithinSC is the call from openScmp? False by default, since it is one of the direct entry points.
+	 */
+	public async openScmpFile(fileUri: vscode.Uri, callFromWithinSC: boolean = false): Promise<void> {
+		if (!callFromWithinSC) {
+			await this.launch(undefined, undefined, false, undefined);			//Instantiate and open schema compare window if called from "Open in Schema Compare"
```
Comment (kisantia) [inline extensions/schema-compare/src/schemaCompareMainWindow.ts:1102]: nit: move the comment to the line above for easier readability and add "command" at the end of the sentence so it's more clear that it's a command

**Review by kisantia — APPROVED** [review]

**Comment by kisantia** [comment]
Looks good! Just want to confirm a couple things:
- schema compare after launching "Open in Schema Compare" from a .scmp file succeeds
- "Open in Schema Compare" only shows for .scmp files and not other file types

File: extensions/schema-compare/src/schemaCompareMainWindow.ts (hunk 3 above)
Comment (SakshiS-harma) [reply extensions/schema-compare/src/schemaCompareMainWindow.ts:1102]: Sure, will do.

**Comment by SakshiS-harma** [comment]
> * n Schema Compare" only shows for .scm

Yes, have tested for both aforementioned scenarios and it works as expected.

**Pull request merged**